bundles/org.eclipse.swt/Eclipse SWT/win32/org/eclipse/swt/graphics/Image.java

```diff
@@ -2256,6 +2360,8 @@ private ImageHandle initializeHandleFromSource(int zoom) {
 
 	protected abstract ElementAtZoom<ImageData> loadImageData(int zoom);
 
+	protected abstract ImageData loadImageData(int targetWidth, int targetHeight);
```

Comment:
What's the difference of `newImageData` and `loadImageData`? Shouldn't that be combined to one method that returns the image data at the appropriate size?

Reply:
well it is also implemented in seperate methods when using a zoom factor instead of a target size. 
`newImageData()` calls `loadImageData()` to get the `ImageData` and afterwards performs a scaling and/or adapts the imageData to look disabled. 

The method `loadImageData()` is overridden in the `ImageDataProviderWrapper` and `ImageFileNameProviderWrapper` to perform the actual loading. 

As stated the existing implementation with zoom factor does the same but nested in methods like `initializeHandleFromSource()`.